lib/Models/Definition/createCombinedModel.ts

```diff
@@ -234,6 +236,15 @@ function createCombinedStratum<T extends ModelTraits>(
   return <StratumFromTraits<T>>(<unknown>result);
 }
 
+function decorate(
+  target: any,
+  decorators: { [id: string]: PropertyDecorator }
+) {
```

Comment:
It seems you've changed this away from using `makeObservable`. Could you explain why?

Reply:
> It seems you've changed this away from using makeObservable. Could you explain why?

Not sure what you meant. The only difference from the other `decorate()` definition is that here we are decorating [a plain object](https://github.com/TerriaJS/terriajs/blob/1a2295507cd064e7d14ba9f9c31d9bfce4bb58f8/lib/Models/Definition/createCombinedModel.ts#L189-L193) instead of a [class prototype](https://github.com/TerriaJS/terriajs/blob/1a2295507cd064e7d14ba9f9c31d9bfce4bb58f8/lib/Models/Definition/addModelStrataView.ts#L58).

Reply:
Actually, good catch. No one is calling `makeObservable` on the [result](https://github.com/TerriaJS/terriajs/blob/1a2295507cd064e7d14ba9f9c31d9bfce4bb58f8/lib/Models/Definition/createCombinedModel.ts#L189-L193). 
So in reality, the trait  wasn't getting memoized.

I'll add a `makeObservable(result)` after decorating it.

Reply:
[Done](https://github.com/TerriaJS/terriajs/blob/6f8a680ba2d1d5f3188c24e9d0ce2586fb573881/lib/Models/Definition/createCombinedModel.ts#L235).